### ISSUE TITLE
Where clause generator don't name some parameters as expected when using standalone or / and methods

### ISSUE THREAD

**Issue by iXo** [mybatis/mybatis-dynamic-sql#38]
Hi,

I wanted to generated where clause, and I used the multiple paramerters sample, where we can tell mybatis how to name the parameter.

The problem is that I need to use this kind of query :
`WhereClauseProvider whereClause =
				where(Record.critRecStart, isBetween(DateTime.now().minusYears(2)).and(DateTime.now()))
				.and(Record.critChannelNumber, isEqualTo(1L), or(Record.critChannelNumber, isEqualTo(10L)))
				.build().render(RenderingStrategy.MYBATIS3, "whereClauseProvider");`

Here, you can see, that I am using a 'standalone' or method call.

The resulted where clause is :

> "where rec_start between #{whereClauseProvider.parameters.p1,jdbcType=TIMESTAMP,typeHandler=com.thalesgroup.vpro.db.typehandlers.DateTimeTypeHandler} and #{whereClauseProvider.parameters.p2,jdbcType=TIMESTAMP,typeHandler=com.thalesgroup.vpro.db.typehandlers.DateTimeTypeHandler} and (channel_number = #{whereClauseProvider.parameters.p3,jdbcType=INTEGER} or channel_number = #{parameters.p4,jdbcType=INTEGER})"

And we can see that the last parameter (p4) is not prefixed with whereClauseProvider, resulting in a query that can't be executed.

**Comment by iXo** [comment]
I think I found where is the problem : 

`private <S> FragmentAndParameters renderSubCriterion(SqlCriterion<S> subCriterion) {
        return CriterionRenderer.withCriterion(subCriterion)
                .withSequence(sequence)
                .withRenderingStrategy(renderingStrategy)
                .withTableAliasCalculator(tableAliasCalculator)
                .build()
                .render();
    }`

In CriterionRenderer, is missing the call to .withParameterName(parameterName) on the renderer builder.

**Comment by jeffgbutler** [comment]
You are correct - great catch!  Feel free to submit a pull request if you want.  I'll fix it later today if you don't.

Thanks!
